fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -3055,7 +3090,14 @@ public boolean isIndexDisabled(@Nonnull String indexName) {
                     Index index = indexItem.getKey();
                     List<RecordType> recordTypes = indexItem.getValue();
                     IndexState indexState = newStates.getOrDefault(index, IndexState.READABLE);
-                    work.add(rebuildOrMarkIndex(index, indexState, recordTypes, reason, oldMetaDataVersion));
+                    final CompletableFuture<Void> rebuildOrMarkIndexSafely = MoreAsyncUtil.handleOnException(
+                            () -> rebuildOrMarkIndex(index, indexState, recordTypes, reason, oldMetaDataVersion),
+                            exception -> {
+                                // If there is anything issue, simply mark the index as disabled without blocking checkVersion
+                                logExceptionAsWarn("unable build index", exception);
```

Comment:
```suggestion
                                logExceptionAsWarn("unable to build index", exception);
```

It also seems like we might want to include things like the index name as a log key. Maybe `logExceptionAsWarn` should take a `KeyValueLogMessage` (and handle adding the log keys from the exception as well)?